src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useCallback, useState } from 'react';
+import debounce from 'lodash.debounce';
+import { peopleFromServer } from '../data/people';
+import { Person } from '../types/Person';
+
+type Props = {
+  setSelectedPerson: (person: Person) => void,
+  delay: number,
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  setSelectedPerson,
+  delay,
+}) => {
+  const [query, setQuery] = useState('');
+  const [isClicked, setIsClicked] = useState(false);
+  const [isMatching, setIsMatching] = useState(true);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, delay),
+    [],
+  );
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setIsMatching(true);
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const handleSearch = (personName: string) => {
+    setQuery(personName);
+    setSelectedPerson(peopleFromServer[
+      peopleFromServer.findIndex(person => person.name === personName)
+    ]);
+    setIsMatching(false);
+  };
+
+  const filteredPeople = (
+    [...peopleFromServer].filter(person => {
```

Comment:
No need to create a copy here 
![image](https://github.com/mate-academy/react_autocomplete/assets/99403198/29d16a50-b819-4f40-846a-67cc2e1c60b9)